mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/proxy/ConstructDeclarationProxy.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal.loader.enricher.proxy;
+
+import org.mule.runtime.api.meta.MuleVersion;
+import org.mule.runtime.api.meta.model.ComponentVisibility;
+import org.mule.runtime.api.meta.model.ModelProperty;
+import org.mule.runtime.api.meta.model.declaration.fluent.*;
+import org.mule.runtime.api.meta.model.deprecated.DeprecationModel;
+import org.mule.runtime.api.meta.model.display.DisplayModel;
+import org.mule.runtime.api.meta.model.error.ErrorModel;
+import org.mule.runtime.api.meta.model.stereotype.StereotypeModel;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Implementation of {@link ConstructDeclaration} that proxies to the {@link OperationDeclaration} that it actually models, after
+ * making the Routers be modeled as Operations instead of Constructs. This is only to be used to ensure Backward Compatibility
+ * with preexistent privilege extensions that have routers, as Aggregators.
+ *
+ * @since 1.7
+ */
+public class ConstructDeclarationProxy extends ConstructDeclaration {
```

Comment:
not technically a proxy and not something that describes what it's doing. Rename to something like `OperationToConstructAdapter`, or `ConstructForwarderDecorator` or something like that.